### DISCUSSION TITLE
CLI: Error passing variables to scale operator

### DISCUSSION THREAD

**Discussion by nelio** [ImageMagick/ImageMagick#5808] [Help] [answered]
I'm having some issues passing variables to the scale operator.
Consider this command line with a 350x500 image

`convert in.jpg \( +clone -crop 1x+0+0 +repage -scale "50000%%x100%%" \) +swap -gravity East -composite out.jpg`

this would essentially pick the letftmost column 1x500 of the image and extend it so we get a 1:1 (500x500) image.

however if I'd want to apply this transformation to any image no matter the geometry (as long as its dimentions are h>w) I thought I'd use something like this:

`convert in.jpg \( +clone -crop 1x+0+0 +repage -scale "%[h]00%%x100%%" \) +swap -gravity East -composite out.jpg`

This however produces an error:
`convert: invalid argument for option '-scale': %[h]00%%x100%% @ error/convert.c/ConvertImageCommand/2720.`

Am I doing something wrong or is it not possible to pass variables to the scale operator in this way?

**Comment by snibgo** [comment, marked as answer]
> ...pass variables to the scale operator in this way?

Only in IM v7 with "magick".

What version of IM, on what platform. "\\(" looks like bash, but you double the %% which looks like Windows BAT.

I can see what you are doing, but "`%[h]00`" isn't a number, and doesn't evaluate to a number.

Instead, I suggest using an "`fx:`" expression. This can only be done in IM v7, with "magick", not "convert". The expression would be `%[fx:h*100]`. So the bash command could be:
```
magick in.jpg \( +clone -crop 1x+0+0 +repage -scale "%[fx:h*100]x100%" \) +swap -gravity East -composite out.jpg
```

**Reply by nelio** [reply]
Brilliant, thank you so much for your help! 
I didn't realise that it needed to be a number, just assumed it was being interpreted as a string, that explains it!
